extensions.bzl

```
@@ -184,56 +231,68 @@ def _doxygen_extension_impl(ctx):
-            "mac": struct(version = "1.12.0", sha256 = "6ace7dde967d41f4e293d034a67eb2c7edd61318491ee3131112173a77344001"),
-            "linux": struct(version = "1.12.0", sha256 = "3c42c3f3fb206732b503862d9c9c11978920a8214f223a3950bbf2520be5f647"),
+            "windows": struct(version = "1.12.0", sha256 = "07f1c92cbbb32816689c725539c0951f92c6371d3d7f66dfa3192cbe88dd3138", executable = ""),
+            "mac": struct(version = "1.12.0", sha256 = "6ace7dde967d41f4e293d034a67eb2c7edd61318491ee3131112173a77344001", executable = ""),
+            "mac-arm": struct(version = "1.12.0", sha256 = "6ace7dde967d41f4e293d034a67eb2c7edd61318491ee3131112173a77344001", executable = ""),
+            "linux": struct(version = "1.12.0", sha256 = "3c42c3f3fb206732b503862d9c9c11978920a8214f223a3950bbf2520be5f647", executable = ""),
+            "linux-arm": struct(version = "1.12.0", sha256 = "3c42c3f3fb206732b503862d9c9c11978920a8214f223a3950bbf2520be5f647", executable = ""),
         }
 
         # Otherwise, add all the configurations (version and sha256) for each platform
-        for attr in mod.tags.version:
-            platform = attr.platform if attr.platform != "" else "windows" if ctx.os.name.startswith("windows") else "mac" if ctx.os.name.startswith("mac") else "linux"
+        for attr in mod.tags.configuration:
+            platform = attr.platform if attr.platform != "" else _get_current_platform(ctx)
+            if attr.version != "" and attr.executable != None:
+                print(attr.version, attr.executable)
+                fail("`Version` and `executable` are mutually exclusive")
+            if attr.version == "" and attr.executable == None:
+                fail("Exactly one between `version` and `executable` must be specified")
             if platform not in default_configurations:
-                fail("Unsupported platform: '%s'. Available options are (windows, mac, linux)" % platform)
+                fail("Unsupported platform: '%s'. Available options are (windows, mac, mac-arm, linux, linux-arm)" % platform)
             if platform in platforms:
-                fail("Doxygen version/sha256 for platform '%s' was already specified: (version = '%s', sha256 = '%s')" % (platform, versions[platforms.index(platform)], sha256s[platforms.index(platform)]))
+                fail("Doxygen configuration for platform '%s' was already specified: (version = '%s', sha256 = '%s', executable = %s)" % (platform, versions[platforms.index(platform)], sha256s[platforms.index(platform)], executables[platforms.index(platform)]))
             platforms.append(platform)
-            versions.append(attr.version if attr.version != "" else fail("Version must be specified"))
+            versions.append(attr.version)
             sha256s.append(attr.sha256 if attr.sha256 != "" else "0" * 64)
+            executables.append(str(ctx.path(attr.executable)) if attr.executable != None else "")
```

This is likely candidate for why `executables` are getting stored as paths in `MODULE.bazel.lock`

Trying out some fixes now...

---

```suggestion
            executables.append(str(attr.executable) if attr.executable != None else "")
```

`MODULE.bazel.lock` looks better after that:

```
         "generatedRepoSpecs": {
@@ -921,10 +916,10 @@
                 "linux-arm"
               ],
               "executables": [
-                "/home/kayce/.cache/bazel/_bazel_kayce/9659373b1552c281136de1c8eeb3080d/external/+_repo_rules+doxybin/ubuntu-gcc",
-                "/home/kayce/.cache/bazel/_bazel_kayce/9659373b1552c281136de1c8eeb3080d/external/+_repo_rules+doxybin/mac-intel",
-                "/home/kayce/.cache/bazel/_bazel_kayce/9659373b1552c281136de1c8eeb3080d/external/+_repo_rules+doxybin/mac-arm",
-                "/home/kayce/.cache/bazel/_bazel_kayce/9659373b1552c281136de1c8eeb3080d/external/+_repo_rules+doxybin/windows.exe",
+                "'@@+_repo_rules+doxybin//:ubuntu-gcc'",
+                "'@@+_repo_rules+doxybin//:mac-intel'",
+                "'@@+_repo_rules+doxybin//:mac-arm'",
+                "'@@+_repo_rules+doxybin//:windows.exe'",
                 ""
               ]
             }
```